### PR TITLE
feat: Add SAM2 to our model zoo

### PR DESCRIPTION
**Pull request by peri044**
# Description

Add Sam2 model to our model zoo

## Type of change

- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project (You can use the linters)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas and hacks
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests to verify my fix or my feature
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

File: examples/dynamo/sam/README.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+Torch-TensorRT.
+
+## Dependencies
+Install the following dependencies before running the model.
+
+```py
+pip install -r ../requirements.txt
+```
+
+Certain custom modifications are required to ensure the model is exported successfully. To apply these changes, please install SAM2 using the following fork : https://github.com/chohk88/sam2/tree/torch-trt (<a href="https://github.com/chohk88/sam2/tree/torch-trt?tab=readme-ov-file#installation">Installation</a>)
```
Comment (narendasan) [inline examples/dynamo/sam/README.md:13]: Can you describe the modifications?

File: examples/dynamo/sam/README.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+## Model execution
+Run the following script to produce output masks after compiling with Torch-TensorRT
+```py
+python run_sam.py
```
Comment (narendasan) [inline examples/dynamo/sam/README.md:18]: Shouldnt this code be part of a single file like the other sphinx-gallery entries?

File: examples/dynamo/sam/README.md (hunk 1 above)
Comment (chohk88) [reply examples/dynamo/sam/README.md:13]: I have added a description of the modifications to the SAMv2 model.

**Review by narendasan — APPROVED** [review]
LGTM

**Review by chohk88 — APPROVED** [review]
I've made the necessary updates to the README. LGTM!

**Pull request merged**